### ISSUE TITLE
meaningless and strange parsing error mesage

### ISSUE THREAD

**Issue by dlucanu** [kframework/k-legacy#1352]
Let's consider the following definition

```
module TEST

  syntax Val ::= Int | Bool

  syntax Exp ::=
       Id
     | Val
     | "(" Exp ")" [bracket]
     > "!" Exp
     | "printf" "(" String ")"

  syntax Ids ::= List{Id, ","}
  syntax Exps ::= List{Exp, ","}
  syntax ExpSeq ::= List{Exp, ";"}

  syntax Decl ::=
       "int" Exps
     | "bool" Exps

  syntax DeclSeq ::= List{Decl, ";"}

  syntax Stmt ::= 
       Decl ";"
     | ExpSeq
     | "proctype" Id "(" DeclSeq ")" "{" Stmts "}"
     | "init" "{" Stmts "}"
     | Exp "->" Stmt

  syntax Stmts ::= List{Stmt, ""}

  configuration
    <k> $PGM:Stmts /* ~> executeInit */ </k>

endmodule
```

and the following program

```
init {
    printf("passed first test!\n")
}
```

The definition compiles, but running the above program I got the following error message:

```
$ kompile test.k 
$ krun ex.test 
Fatal error at
    javaParseStringPgm_0_0
    parse_string_1_2
    parse_string_1_3
    jsglr_parse_string_1_3
[Error] Internal: Uncaught exception thrown of type ProvisionException.
Please file a bug report at https://github.com/kframework/k/issues
```

**Comment by dlucanu** [comment]
I made more tests and it seems that the source of the error is the syntax declaration

```
syntax Stmt ::= ExpSeq
```

If is replaced by 

```
syntax Stmt ::= Exp
```

then krun works as expected.

**Comment by radumereuta** [comment]
I tried the example and run `kast pgm.test --debug` and managed to reproduce it.
Unfortunately the error I got is as follows:

```
at org.spoofax.jsglr.client.Disambiguator.filterAmbiguities(Disambiguator.java:820)
at org.spoofax.jsglr.client.Disambiguator.filterTree(Disambiguator.java:469)
```

Indicating a bug in SDF. It's not the first time I see something like this, and it's not something that I can fix.
I will leave this issue open and test to see if it's still an issue in the new parser (most likely it won't).
